src/PCN.py

```diff
@@ -0,0 +1,138 @@
+'''
+Base class for PCNs. 
+
+'''
+import networkx as nx
+import MDAnalysis as mda
+import numpy as np
+import random
+
+class PCN():
+	'''
+	Base class for PCNs. 
+	
+	Takes a PDB file and creates a PCN from the alpha carbons.
+
+	'''
+	# Define a threshold value for defining a link between two atoms
+	link_length_threshold = 8.0 # Å
+	
+	def __init__(self, PDB_file):
+		self.universe = mda.Universe(PDB_file)
+
+	def get_C_alphas(self):
+		'''
+		Take a universe, get segment A and return the C-alphas for the PDB. 
+
+		Parameters
+		----------
+		universe: MDAnalysis.Universe(PDB_file)
+
+		Return
+		------
+		C_alphas: MDAnalysis AtomGroup
+			contains the C-alphas from the PDB
+		'''
+		"------------------------MOVE-------------------------------"
+		segments = self.universe.residues.segments
+			
+		number_of_segments = len(segments)
+
+		for index in range(number_of_segments):
+			segment_id = self.universe.residues.segments.segids[index]
+
+			try:
+				C_alphas = self.universe.select_atoms(f'name CA and segid {segment_id}')
+				#print(segment_id)
+				break
+			
+			except:
+				print('Error in first segment id. Trying the next one.')
+				continue
+
+		"-----------------------------------------------------------"
+		return C_alphas
+
+	def get_chain_length(self,C_alphas):
+		'''
+		Take C-alphas and use that to return the length of the chain.
+
+		Parameters
+		----------
+		C_alphas: MDAnalysis AtomGroup
+			contains the C-alphas from the PDB
+
+		Return
+		------
+		chain_length: int
+			length of the chain
+		'''
+		chain_length = len(C_alphas)
+		
+		return chain_length
+
+	def create_connected_component_subgraphs(self,protein_graph):
```

Comment:
make sure this is formatted in the same way self, protein_graph. I would recommend getting something like pycharm and let it autoformat everything according to pep8

Reply:
Yep fixed.